### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by Rondom**
What I plan to do (feel free to comment/discuss):
* "Introduction" (remove it)
* "Features"
  * adapt language, remove outdated info and features that are not features
  * maybe add one or two sentences at the top of the README describing what libdumb does
* Remove any info on how to build. Instead, point to COMPILING.md.
* "Examples"
   * update
   * Should we move this information into a separate readme the examples folder?
* "Downloading music or writing your own"
   * The information is "interesting" but does not belong into the readme IMO. Do you think it is worth keeping? Obviously it also needs some updating (e.g. point to the @kode54's foobar2k plugin instead of WinAmp...)
* "Contact details"
   * replace with GitHub URL and invitation to submit bugs and PRs
* "Conclusion"
  * remove, obviously

### CODE REVIEW

**Comment by kode54** [comment]
Do continue with this, I’ll track it.

**Comment by katajakasa** [comment]
A suggestion for base:

## 1. Introduction

DUMB is a module audio renderer library. It reads your mod/it/etc. files and
outputs audio that you can dump to your actual playback library.

## 2. Features

The following formats are supported:
* IT
* 669
* AMF
* ASY
* MOD
* MTM
* OKT
* PSM (old and new)
* PTM
* RIFF
* S3M
* STM
* XM

Other features:
* Seeking
* Selectable output rate
* Selectable resampling quality
* SSE optimizations
* Optional Allegro4 integration support
* <add more here>

## 3. Installation

Currently you need to compile libdumb yourself. For more details, please see
file COMPILING.md.

## 4. License

Please see LICENSE for license details.

**Comment by katajakasa** [comment]
* "Contact details" => perhaps a separate CONTRIBUTING.md ?
* "Introduction" => I'd keep a short explanation on what the library is and what it does (see above)
* ."Downloading music or writing your own" ... maybe drop ? And add places where libdumb is used to introduction ?

**Comment by Rondom** [comment]
Everyone: Let me know, about any feedback. As it stands, I consider at least this PR ready for merging if no-one objects. One can always submit your own PRs with smaller-step improvements later.

@katajakasa 
> "Contact details" => perhaps a separate CONTRIBUTING.md ?

I considered it, but I would say once we have more than 5 lines of text, we can do that. :-)

> "Downloading music or writing your own" ... maybe drop ?

The information is outdated and needs updating. Without some knowledge about the demoscene and the different tracker software used nowadays I do not think I could update it and removed it.

> And add places where libdumb is used to theintroduction ?

Good idea, but I would only put in prominent examples, though. (which?)

**Comment by SimonN** [comment]
Looks good to me! Good choice of what to keep in the bulleted list. Explaining the niche formats in a longer list (like you did in commit 20898ae) is deserved, it's the main library feature. You're still pointing to the build documentation in due time, and the build docs will be in a standard place.

Prominent uses: I don't know any others besides Allegro 4 and Allegro 5, which seem the most prominent.

I've indeed stumbled on this fork because A5 doesn't support DUMB 1.0 and I'm working towards that. :-) I'm documenting the `DUMBFILE_SYSTEM` and will open a PR this weekend.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Two simple example programs are provided.
+
+
+## dumbplay
+
+dumplay will play a module file from command-line. It requires SDL2 for audio
+output and argtable2 for argument parsing.
+
+
+### dumbout
```
Comment (katajakasa) [inline examples/README.md]: ## instead of ### ?

Comment (Rondom) [reply examples/README.md]: thx, fixed

**Comment by katajakasa** [comment]
Well, I think it's good. Short, up-to-date and to the point.

**Comment by katajakasa** [comment]
If you wish to add well known usage examples, then at least DeadBeef and foobar2000 plugin could be mentioned.

**Comment by Rondom** [comment]
Can someone double-check the feature-list for factual accuracy? Is it complete or are things missing?
About the format: Is AMF and ASY the same thing? What is the RIFF-format? Where has it been used? To me it sounds like a container for other formats.

I will leave the usage examples to someone else.

**Comment by kode54** [comment]
Asylum is one of the two formats DUMB supports that uses the .AMF extension. It was also the first of the two that I added support for.

The RIFF parser is used for three formats: DSMF, and AM/AMFF, which are two versions of the same format, yet wildly different in general file structure. See: readriff.c, readam.c, and readdsmf.c.

DSMF is a slightly obscure format used by some old games.

AM and AMFF are the core formats used by Jazz Jackrabbit 2, and possibly other games. The description for the compressed container they are packaged in may be located in mod.cpp in foo_dumb, but it's really just a CRC32 protected zlib deflate block containing the RIFF AM or AMFF structure.

**Comment by Rondom** [comment]
@kode54 Thanks for the information. Maybe you can just merge this PR and do any corrections yourself, because, frankly, even though your explanation was good, I am not sure what to write about the different formats, now. Thus it would be probably be more efficient for you to edit the bulleted list yourself and correct any inaccuracies.

**Comment by SimonN** [comment]
@kode54, please merge this documentation soon. This allows future PRs to improve the docs even more.

**Comment by katajakasa** [comment]
Maybe one more thing: Add a nod to the original developer(s) in the readme introduction, and mention the original website.

**Pull request merged**